main/read.c

```diff
@@ -592,6 +593,36 @@ extern MIO *getMio (const char *const fileName, const char *const openMode,
 	return mio_new_memory (data, size, eRealloc, eFree);
 }
 
+/* Return true if utf8 BOM is found */
+static bool checkUTF8BOM (MIO *mio, bool skipIfFound)
+{
+	bool r = false;
+	if ((0xEF == mio_getc (mio))
+		&& (0xBB == mio_getc (mio))
+		&& (0xBF == mio_getc (mio)))
+		r = true;
+
+	if (r && skipIfFound)
```

Comment:
You are correct!
Terribly bad. I wonder why the test case cannot detect this mistake.

...

rewindInputFile is called via resetInputFile.
resetInputFile is called from createTagsForFile.
This means we can remove duplicated code in rewindInputFile and openInputFile from openInputFile...